effekt/shared/src/main/scala/effekt/generator/llvm/Transformer.scala

```diff
@@ -36,7 +35,8 @@ object Transformer {
       val instructions = BC.instructions; BC.instructions = null;
 
       val entryBlock = BasicBlock("entry", instructions, terminator)
-      val entryFunction = Function(VoidType(), "effektMain", List(), entryBlock :: basicBlocks)
+      // TODO strictly speaking, the entry function should use the C calling convention
+      val entryFunction = Function(Tailcc(), VoidType(), "effektMain", List(), entryBlock :: basicBlocks)
```

Comment:
I don't understand why it should use the `C` calling convention, could you explain? Isn't this a "jump" to main?

Reply:
For documentation: This one is literally called from C. It does a tail call to the user-defined main entry point with the tail calling convention and musttail. When trying to give it the C calling convention this musttail can not be guaranteed.